services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/dao/GreRangeRepository.java

```diff
@@ -121,7 +122,9 @@ public synchronized Long allocateGreKey (String rangeId) throws Exception {
             }
 
             key = networkGRERange.allocateKey();
-            cache.put(networkGRERange.getId(), networkGRERange);
+            if (!key.equals(ConstantsConfig.keyNotEnoughReturnValue)) {
```

Comment:
Recommend to put this check into a util method which could be used for other extended validation, not only checking keyNotEnoughReturnValue.

Reply:
Yeah that's right ~ but we could make this change when we have other extended validation later, what do you think

Reply:
Sure.